inc/StyleUtils.php

```diff
@@ -4,101 +4,161 @@
+     * @param string $tpl template name
+     * @param bool $preview
+     * @param bool $reinit whether static style conf should be reinitialized
+     */
+    public function __construct($tpl = '', $preview = false, $reinit = false)
+    {
+        if (!$tpl) {
+            global $conf;
+            $tpl = $conf['conf'];
+        }
+        $this->tpl = $tpl;
+        $this->reinit = $reinit;
+        $this->preview = $preview;
+    }
+
     /**
      * Load style ini contents
      *
      * Loads and merges style.ini files from template and config and prepares
      * the stylesheet modes
      *
      * @author Andreas Gohr <andi@splitbrain.org>
+     * @author Anna Dabrowska <info@cosmocode.de>
      *
-     * @param string $tpl the used template
-     * @param bool   $preview load preview replacements
      * @return array with keys 'stylesheets' and 'replacements'
      */
-    public function cssStyleini($tpl, $preview=false) {
+    public function cssStyleini() {
         global $conf;
 
-        $stylesheets = array(); // mode, file => base
-        // guaranteed placeholder => value
-        $replacements = array(
-            '__text__' => "#000",
-            '__background__' => "#fff",
-            '__text_alt__' => "#999",
-            '__background_alt__' => "#eee",
-            '__text_neu__' => "#666",
-            '__background_neu__' => "#ddd",
-            '__border__' => "#ccc",
-            '__highlight__' => "#ff9",
-            '__link__' => "#00f",
+        $mergedinis = $this->mergeStyleInis();
+
+        return array(
+            'stylesheets' => $mergedinis['stylesheets'],
+            'replacements' => $mergedinis['replacements']
         );
+    }
 
-        // load template's style.ini
-        $incbase = tpl_incdir($tpl);
-        $webbase = tpl_basedir($tpl);
-        $ini = $incbase.'style.ini';
-        if(file_exists($ini)){
-            $data = parse_ini_file($ini, true);
-
-            // stylesheets
-            if(is_array($data['stylesheets'])) foreach($data['stylesheets'] as $file => $mode){
-                if (!file_exists($incbase . $file)) {
-                    list($extension, $basename) = array_map('strrev', explode('.', strrev($file), 2));
-                    $newExtension = $extension === 'css' ? 'less' : 'css';
-                    if (file_exists($incbase . $basename . '.' . $newExtension)) {
-                        $stylesheets[$mode][$incbase . $basename . '.' . $newExtension] = $webbase;
-                        if ($conf['allowdebug']) {
-                            msg("Stylesheet $file not found, using $basename.$newExtension instead. Please contact developer of \"{$conf['template']}\" template.", 2);
-                        }
-                        continue;
-                    }
-                }
-                $stylesheets[$mode][$incbase . $file] = $webbase;
+    /**
+     * @return array
```

Comment:
The return format should be explained here and a short info what the method does be added.